src/generator/index.ts

```diff
@@ -69,3 +72,7 @@ export class MonoClientGenerator {
     }
   }
 }
+
+function isServiceIdExtended(object: any): object is ServiceIdExtended {
+  return typeof object != 'string' && typeof object != 'number' && 'serviceId' in object;
```

Comment:
If you change serviceId as optional you can do this
```suggestion
  return typeof object === 'object' && object != null;
```

Reply:
done!